### PR TITLE
terraform-providers: misc updates

### PR DESCRIPTION
**Pull request by zowoq**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/cluster/terraform-providers/providers.json [hunk 1]
```diff
@@ -188,10 +188,12 @@
     "version": "2.1.2"
   },
   "cloudstack": {
-    "owner": "terraform-providers",
+    "owner": "cloudstack",
+    "provider-source-address_broken": "registry.terraform.io/cloudstack/cloudstack",
```
Comment (zowoq) [inline pkgs/applications/networking/cluster/terraform-providers/providers.json:192]: The version on the registry doesn't have a corresponding tag in the source repo so doing this just so it's skipped by the update script.

File: pkgs/applications/networking/cluster/terraform-providers/providers.json [hunk 2]
```diff
@@ -824,9 +826,11 @@
   },
   "segment": {
     "owner": "ajbosco",
+    "provider-source-address_no_registry": null,
```
Comment (zimbatm) [inline pkgs/applications/networking/cluster/terraform-providers/providers.json:829]: what's that?

Comment (zowoq) [inline pkgs/applications/networking/cluster/terraform-providers/providers.json:829]: Skipping the update script here as well, or we could move it to a separate derivation or remove it entirely.

File: pkgs/applications/networking/cluster/terraform-providers/providers.json [hunk 3]
```diff
@@ -1015,13 +1028,6 @@
     "vendorSha256": null,
     "version": "2.0.2"
   },
-  "vthunder": {
```
Comment (zowoq) [inline pkgs/applications/networking/cluster/terraform-providers/providers.json:1018]: It still exists but it redirects to `thunder` and we'd need to override the build to get it to work as `vthunder`.

File: pkgs/applications/networking/cluster/terraform-providers/providers.json (hunk 2 above)
Comment (zimbatm) [reply pkgs/applications/networking/cluster/terraform-providers/providers.json:829]: Let's remove it.

File: pkgs/applications/networking/cluster/terraform-providers/providers.json (hunk 1 above)
Comment (zimbatm) [reply pkgs/applications/networking/cluster/terraform-providers/providers.json:192]: Let's remove all the broken packages.

**Comment by zimbatm** [comment]
By the way, I don't know if you've seen https://github.com/numtide/nixpkgs-terraform-providers . It takes a bit of a different stance as it uses binary releases. The repo is about auto-updating the providers as I find often the case that I want to bump the terraform providers independently from the nixpkgs pin. The main downside (for now) is that the $out filesystem structure is not compatible with the ones from nixpkgs so you can't mix-and-match. With terraform 1.0 and only registry providers, it will become possible to converge both.

**Comment by zowoq** [comment]
> https://github.com/numtide/nixpkgs-terraform-providers 

No, I hadn't seen that. Having separate pinning looks useful. As it's only using binary releases have you thought about importing the entire registry automatically?

**Comment by zowoq** [comment]
We should look at switching to the `$owner.$provider` namespacing here, it would make dealing with forked/replacement providers much easier.

**Comment by zimbatm** [comment]
> As it's only using binary releases have you thought about importing the entire registry automatically?

I did a manual import of the top official providers, but automating that sounds good.

**Review by zimbatm — APPROVED** [review]

**Comment by zimbatm** [comment]
Result of `nixpkgs-review pr 153206` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>terraform-full</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>terraform-providers.thunder</li>
  </ul>
</details>

**Pull request merged**